Sources/Build/BuildPlan.swift

```diff
@@ -1370,24 +1370,82 @@ public final class MixedTargetBuildDescription {
-                name: buildArtifactDirectory.pathString,
+                name: buildArtifactDirectory.nativePathString(escaped: false),
                 contents: [
                     VFSOverlay.File(
                         name: "module.modulemap",
                         externalContents:
-                            buildArtifactDirectory.appending(component: "module.modulemap").pathString
+                            buildArtifactDirectory.appending(component: "unextended-module.modulemap").nativePathString(escaped: false)
                     )
                 ]
             )
         ]).write(to: unextendedModuleOverlayPath, fileSystem: fileSystem)
 
+        // all-product-headers.yaml
+        let allProductHeadersPath = buildArtifactDirectory
+            .appending(component: "all-product-headers.yaml")
+
+        try VFSOverlay(roots: [
+            VFSOverlay.Directory(
+                name: clangTargetBuildDescription.clangTarget.includeDir.pathString,
+                contents:
+                    // TODO(ncooke3): Why is clangTargetBuildDescription.clangTarget.headers empty?
+                    try Set(fileSystem.getDirectoryContents(clangTargetBuildDescription.clangTarget.includeDir)
+                    .map(clangTargetBuildDescription.clangTarget.includeDir.appending(component:)))
+                        .filter { headerPath in
+                            headerPath.pathString.hasPrefix(clangTargetBuildDescription.clangTarget.includeDir.pathString)
+                        }.map { headerPath in
+                            VFSOverlay.File(
+                                name: headerPath.basename,
+                                externalContents: headerPath.nativePathString(escaped: false)
+                            )
+                        }
+            ),
+            VFSOverlay.Directory(
+                name: buildArtifactDirectory.nativePathString(escaped: false),
+                contents: [
+                    VFSOverlay.File(
+                        name: "module.modulemap",
+                        externalContents:
+                            buildArtifactDirectory.appending(component: "module.modulemap").nativePathString(escaped: false)
+                    )
+                ]
+            ),
+            VFSOverlay.Directory(
+                name: buildArtifactDirectory.nativePathString(escaped: false),
+                contents: [
+                    VFSOverlay.File(
+                        name: "\(target.c99name)-Swift.h",
+                        externalContents: buildArtifactDirectory.appending(component: "\(target.c99name)-Swift.h").nativePathString(escaped: false)
+                    )
+                ]
+            ),
+        ]).write(to: allProductHeadersPath, fileSystem: fileSystem)
+
+        /// unextended-module.modulemap
+        // TODO(ncooke3): The umbrella header is hardcoded. The below block needs
+        // to probabaly move into the module map generation phase.
```

Comment:
probably

Reply:
This comment seems to be referencing an old commit. I double checked and this code does not exist anymore.